ctapipe/irf/irf_classes.py

```diff
@@ -0,0 +1,335 @@
+"""
+Define a parent IrfTool class to hold all the options
+"""
+import astropy.units as u
+import numpy as np
+from astropy.table import QTable
+from pyirf.binning import create_bins_per_decade
+
+from ..core import Component, QualityQuery
+from ..core.traits import Float, Integer, List, Unicode
+
+
+class CutOptimising(Component):
```

Comment:
We don't use components to just store configuration. 

They should *actually do* the thing they are created for. 

E.g. I think it's an anti-pattern to have a class `CutOptimisation` that only holds the config but then the actual optimization happens in the tool.

The class should have  a method to do the cut optimization using it's own traitlets and an input (e.g. signal and background tables)

Reply:
Ok, moved code into it. But the pyirf api looks written for the case when everything is in a shared global scope, which is the opposite of putting things into separate containers.

Reply:
This should be renamed `CutOptimizer` to be consistent with other Components (they are named as nouns by the thing they do, e.g. "EnergyRegressor", "ImageCleaner")

Reply:
renamed